docs/Parameters.rst

```diff
@@ -514,6 +514,12 @@ Learning Control Parameters
 
    -  you need to specify all features in order. For example, ``mc=-1,0,1`` means decreasing for 1st feature, non-constraint for 2nd feature and increasing for the 3rd feature
 
+   -  in the CLI or C++, use a string like ``"-1,0,1"``
+
+   -  in the Python package, can use either a string or a list like ``[-1, 0, 1]``
```

Comment:
@jameslamb I'd like to keep this important doc in s consistent state. Changes proposed in this this PR are actually applicable for any `multi-***`-type parameter. We have a lot of such params, e.g. `max_bin_by_feature `, `cegb_penalty_feature_lazy`, `cegb_penalty_feature_coupled`, `categorical_feature`, `label_gain`, etc. I think it will be better to write a separate paragraph about how `multi-***` params can be passed to a program, if you think there should be some clarification for this.

Also, I'm against documenting internal string format for language wrappers. Actually, all params are passed via a string internally.
https://github.com/microsoft/LightGBM/pull/4101#discussion_r639267831
I don't think we should expose this and it's better to encourage users to use native language structures to pass params.

Reply:
ok, I agree with both of these points. I'll split the unit tests part of this into a separate PR.

> if you think there should be some clarification for this.

I definitely do. If an expert LightGBM user like @mayer79 wasn't aware (https://github.com/microsoft/LightGBM/issues/4345#issuecomment-855659189) then I think many others will be unaware or will spend time trying to figure it out from unit tests / example code.

Reply:
Ok, I've changed this to just document the concept generally. I think it's ready for review.

![image](https://user-images.githubusercontent.com/7608904/121109388-14716480-c7d1-11eb-8769-978fa0931a84.png)